internal/resources/ekscluster/resource_ekscluster.go

```diff
@@ -319,6 +317,38 @@ func resourceClusterDelete(_ context.Context, d *schema.ResourceData, m interfac
 	return diags
 }
 
+func resourceClusterInPlaceUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config := m.(authctx.TanzuContext)
+
+	// Get call to initialise the cluster struct
+	getResp, err := config.TMCConnection.EKSClusterResourceService.EksClusterResourceServiceGet(constructFullname(d))
+	if err != nil {
+		return diag.FromErr(errors.Wrapf(err, "Unable to get Tanzu Mission Control EKS cluster entry, name : %s", d.Get(NameKey)))
+	}
+
+	clusterSpec := constructSpec(d)
+
+	// TODO: update nodepools seperatly
+	clusterSpec.NodePools = nil
+
+	getResp.EksCluster.Meta = common.ConstructMeta(d)
+	getResp.EksCluster.Spec = clusterSpec
+
+	_, err = config.TMCConnection.EKSClusterResourceService.EksClusterResourceServiceUpdate(
```

Comment:
Pretty the whole resource can be updated ?

Reply:
Yes, we can update the cluster config and meta in a single request.